pkg/server/container_create_windows.go

```diff
@@ -166,8 +171,28 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 
 	log.G(ctx).Debugf("Container %q spec: %#+v", id, spew.NewFormatter(spec))
 
-	snapshotterOpt := snapshots.WithLabels(config.Annotations)
+	// If the config field is specified, set the snapshotter label to reuse the pods
+	// scratch space. This only affects LCOW at the moment.
+	val, ok := config.Annotations["containerd.io/snapshot/io.microsoft.container.storage.reuse-scratch"]
+	if rhcso.ShareScratch || (ok && val == "true") {
+		config.Annotations["containerd.io/snapshot/io.microsoft.container.storage.reuse-scratch"] = "true"
+		_, ok := config.Annotations["containerd.io/snapshot/io.microsoft.owner.key"]
```

Comment:
Sorry dont mean to jump in as I have not been looking at any of these but this caught my eye. Are you allowing the caller to pick a different scratch owner than its pod? I'm surprised we dont have the opposite logic "if ok, fail or overwrite it"

Reply:
Please continue to jump in 😆. So the annotation is definitely there, but from the callers perspective we just want them to be able to just specify reuse-scratch and if via cri it will just share the sandbox containers scratch, but I do leave it open to possibly support sharing a specific containers instead.

Reply:
So at a minimum if we want to enable that I think we need to verify that a specific id is part of this pod. I'm concerned that this could inadvertently cause us to mount a vhd cross-uvm or something odd

Reply:
Yea 100%, I didn't even document this in the wiki I wrote up for this. I was trying to find a nice abstraction for this at the snapshotter layer already as there's no pod concept, it's just up to the client to provide the right labels and know they're in the same pod (VM).

Reply:
That makes sense. Maybe you should take in the snapshots defined as some other annotation rather than the snapshotter ones. I guess it doesnt really matter. If we ever allow passthrough annotations on our end we will have to explicitly make sure no customer can pass this one unless its handled here.